day35/reverseStr.py

```diff
@@ -0,0 +1,60 @@
+def reverseStr(s, k):
+    # edge
+    size = len(s)
+    if size < k:
+        return s[::-1]
+
+    # convert string to list
+    letters = list(s)
+    i = 0
+
+    # reverse all the full chunks
+    while i + 2*k-1 < size:
+        reverse(letters, i, i+k-1)
+        i += 2*k
+
+    # take care of the remainder
+    if i+k-1 < size:
+        reverse(letters, i, i+k-1)
+    else:
+        reverse(letters, i, size-1)
+
+    # fin
+    return "".join(letters)
+
+# reverses the indices of arr between start and end, in place
+def reverse(arr, start, end):
+    while start < end:
+        arr[start], arr[end] = arr[end], arr[start]
```

Comment:
hilariously there's actually a hidden allocation here that you might not see. you make a tuple and immediately break it down into two variables